packages/expo-notifications/src/NotificationsHandler.ts

```diff
@@ -0,0 +1,69 @@
+import { EventEmitter, Subscription, CodedError } from '@unimodules/core';
+import NotificationsHandlerModule, { NotificationBehavior } from './NotificationsHandlerModule';
+
+type Notification = any;
+
+export class NotificationTimeoutError extends CodedError {
+  info: { notification: Notification; id: string };
+  constructor(notificationId: string, notification: Notification) {
+    super('ERR_NOTIFICATION_TIMEOUT', `Notification handling timed out for ID ${notificationId}.`);
+    this.info = { id: notificationId, notification };
+  }
+}
+
+export type NotificationHandlingError = NotificationTimeoutError | Error;
+
+export interface NotificationDelegate {
+  handleNotification: (notification: Notification) => Promise<NotificationBehavior>;
+  handleSuccess?: (notificationId: string) => void;
+  handleError?: (error: NotificationHandlingError) => void;
+}
+
+type HandleNotificationEvent = {
+  id: string;
+  notification: Notification;
+};
+
+type HandleNotificationTimeoutEvent = HandleNotificationEvent;
+
+// Web uses SyntheticEventEmitter
+const notificationEmitter = new EventEmitter(NotificationsHandlerModule);
+
+const handleNotificationEventName = 'onHandleNotification';
+const handleNotificationTimeoutEventName = 'onHandleNotificationTimeout';
+
+let handleSubscription: Subscription | null = null;
+let handleTimeoutSubscription: Subscription | null = null;
+
+export function setNotificationDelegate(delegate: NotificationDelegate): void {
```

Comment:
can we call this `setNotificationHandler` (with param `handler: NotificationHandler`) rather than `Delegate`? i like that it's more consistent with the module name and i think is also more idiomatic to JS.